src/task.c

```diff
@@ -320,7 +328,15 @@ task_create_thread_with_hooks(platform_thread       *thread,
                               platform_heap_id       hid)
 {
    platform_status ret;
-   thread_invoke  *thread_to_create = TYPED_ZALLOC(hid, thread_to_create);
+   uint64         *tid_bitmask = task_system_get_tid_bitmask(ts);
+   if (*tid_bitmask == 0) {
+      platform_error_log("Cannot create a new thread as the limit on"
```

Comment:
@rtjohnso - I fully realize that this is racy. But trying to rework this aspect also in this enablement PR will bloat the scope. The error checks I've added now certainly improve the situation. I think it will be better to peel this off as a follow-on issue. 

What if I open a new issue to track this item?